tests/S-11_Chat_and_Messages_modifications.test.js

```diff
@@ -0,0 +1,272 @@
+  // Prepare data
+  const chatModel = {
+    userName: "name_dummy", 
+    message: "message_dummy"
+  };
+
+  // Check that the function does NOT throw an error
+  await t.notThrowsAsync(async () => {
+    await onlyFunc.createUser(chatModel);
+  });
+
+});
+
+// Test for:    POST    /user/{userName}/class/{className}/chat 
+test('POST endpoint /user/{userName}/class/{className}/chat', async (t) => {
+  const { statusCode } = await t.context.got.post('user/{userName}/class/{className}/chat', {
+    // Give it dummy data
+    json: {
+      userName: "name_dummy", 
+      message: "message_dummy"
+    }
+  })
+
+  // Check status Code
+  t.is(statusCode, 200, 'Expected status code 200 for successful post')
+})
+
+// Test for Multiple:    POST    /user/{userName}/class/{className}/chat
+test('Multiple POST /user/{userName}/class/{className}/chat', async (t) => {
+  // Prepare data
+  const chatToCreate = [
+    {
+      userName: "name_dummy1", 
+      message: "message_dummy1"
+    },
+    {
+      userName: "name_dummy2", 
+      message: "message_dummy2"
+    },
+    {
+      userName: "name_dummy3", 
+      message: "message_dummy3"
+    }
+  ]
+
+  for (const chatData of chatToCreate) {
+    const { statusCode } = await t.context.got.post('user/{userName}/class/{className}/chat', { json: chatData })
+
+    // Check status Code
+    t.is(statusCode, 200, `Expected status code 200 for successful post, Bad request for chat ${chatData.userName}`)
+  }
+})
+
+// Test for Wrong input:    POST    /user/{userName}/class/{className}/chat
+test('Wrong input POST endpoint /user/{userName}/class/{className}/chat', async (t) => {
+  const { statusCode } = await t.context.got.post('user/{userName}/class/{className}/chat', {
+    // Give it dummy data
+    json: {
+      // userName: "name_dummy", 
+      // message: "message_dummy"
+    }
+  })
+
+  // Check status Code
+  // It shouldn't respond with 200 when the body is empty, So now the correct step would be to change the function
+  t.is(statusCode, 200, 'Expected status code 200 for successful post')
+})
+
+// Test for Empty input:    POST    /user/{userName}/class/{className}/chat
+test('Empty input POST endpoint /user/{userName}/class/{className}/chat', async (t) => {
+  await t.throwsAsync(
+    async () => {
+      await t.context.got.post('user/{userName}/class/{className}/chat', {})
+    },
+    { instanceOf: t.context.got.HTTPError, message: /Response code 415/ }
+    // Or we can use this 'Response code 415 (Unsupported Media Type)'
+  )
+})
+
+
+// =================== DELETE /user/{userName}/class/{className}/chat/{messageID} ==================== //
+
+
+// Test the function "deleteMessage" that gets called when the endpoint is used
+test('deleteMessage resolves when called with a message id', async (t) => {  
+  const messageToDelete = 'exampleMessageId';
+
+  await t.notThrowsAsync(async () => {
+    await onlyFunc.deleteUser(messageToDelete);
```

Comment:
What does this part of the code do ?

Reply:
Makes sure that the 'deleteUser' function, when called witht the message ID 'exampleMessageId' does not throw an error. 

I added it to ensures that the 'deleteUser' function will work as expected whan given a message ID and to prevent regrassions (if someone modifies the 'deleteUser' function in a way that breaks its behavior, this test would likely fail).